### DISCUSSION TITLE
Is there any interest in rewriting the frontend app?

### DISCUSSION THREAD

**Discussion by mcapu** [chilli-axe/mpc-autofill#130] [Ideas]
Hi,

Historically, I've seen some improvements that I believe could be made to the current UI, such as:

* Reordering slots
* Duplicating slots
* Zooming in on cards while picking a version without needing to select it and then go back to the "list" view
* Filtering by creator/drive without the need to reload the page
* Saving state between sessions
* Undoing (at least for certain operations, such as slot deletion)

Aside from this, the current code is written in plain JS with `jQuery` and it feels a bit harder to add features to it than to rearchitect it from scratch. After taking a quick look at it, I was thinking about writing my own implementation for personal usage. However, it might be a better idea if we could align on this and get to a shared solution to which I could contribute and that could be used by more people. Please let me know if you'd be open to start such a project or if you'd prefer to keep it as-is.

PS: it might be a good idea to open up GitHub discussions for this project. This kind of issues make more sense as discussions to me, but I had no choice.
PS2: great project, thanks for all the effort.

**Comment by mcapu** [comment]
Actually, I just found the `reachitect-django-project` branch, so I'll take a look at it there.

**Comment by ndepaola** [comment]
hey, thanks for posting this :) i agree the current frontend isn't ideal and i've made some progress towards rewriting it in react. it's promising so far but still a fair amount more work required, and i'm finding it hard to put time towards it between full time work and health issues atm. i'll look into moving my project management over to github and opening the discussions board for better transparency here :)

**Comment by ndepaola** [comment]
forgot to set up GitHub discussions until now whoops 😅 it was enabled in the repo settings but I didn't realise I needed to create the first post as well

**Comment by ndepaola** [comment]
initial PR is up here https://github.com/chilli-axe/mpc-autofill/pull/132 going to merge this in now-soonish so I can start testing this against a larger database on a deployed backend. still much more work to do before this is fully ready to use

**Comment by dev0T** [comment]
Came here to write the same. I've be glad to help with the implementation of those features. I will be looking into what @ndepaola linked here and see where I can start contributing in the near future

**Comment by ndepaola** [comment]
awesome!! work on the new frontend is ticking along and i've deployed it (in alpha state) at https://mpcautofill.github.io

the project status is reasonably up-to-date here https://github.com/users/chilli-axe/projects/1 and i've included some docs on how to set up your dev environment for the frontend here https://github.com/chilli-axe/mpc-autofill/blob/master/frontend/readme.md

let me know if you get around to looking through the codebase and we can go from there!

**Comment by ndepaola** [comment]
an update on this - the new frontend is at feature parity + has a slew of new features compared to the old frontend, and the codebase has become reasonably mature, so im considering this work to be in open beta now - getting people hands-on in it is blocked by mpcfill.com pulling my changes from the last few months - still at https://mpcautofill.github.io :)

**Comment by ndepaola** [comment]
the beta for the rewritten frontend is now fully online. check it out with the mpcfill.com database at https://mpcautofill.github.io/?server=mpcfill.com
